types/src/contract_event.rs

```diff
@@ -93,6 +129,16 @@ pub struct ContractEventV0 {
     event_data: Vec<u8>,
 }
 
+/// Entry produced via a call to the `emit` builtin.
+#[derive(Hash, Clone, Eq, PartialEq, Serialize, Deserialize, CryptoHasher)]
+pub struct ContractEventV1 {
+    /// The type of the data
+    type_tag: TypeTag,
+    /// The data payload of the event
+    #[serde(with = "serde_bytes")]
+    event_data: Vec<u8>,
+}
```

Comment:
nit: I think it's slightly better to move this declaration down to right before its own implementation.